### PR TITLE
Allow vm_facts to be restricted to a given folder or cluster in a dat…

### PR DESCRIPTION
**Pull request by ritzk**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

cloud/vmware/vmware_vm_facts.py
##### Summary:

Allows us to restrict vm_facts listing to a particular datacenter ( cluster or folder )
Listing facts for all the VMs is a very slow process, and often not needed.

blocked_by: https://github.com/ansible/ansible/pull/15053
##### Example:

```
---  
    - name: Gather all registered virtual machines
      local_action:
        module: vmware_vm_facts
        hostname: "{{ vmware_exsi_server }}"
        username: "{{ user_name }}"
        validate_certs: false
        datacenter: "{{ datacenter }}"
      register: vm_list

    - debug: msg="{{ vm_list }}"

```

…

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @ritzk. To the current maintainers, @jcpowermac, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: cloud/vmware/vmware_vm_facts.py [hunk 1]
```diff
@@ -51,41 +81,93 @@
...
+            # list all VMs inside a folder aka "Templates and folder"
+            elif folder:
+                folder = find_entity_child_by_path(content, datacenter.vmFolder,folder)
+                _virtual_machines = get_all_virtual_machines(content, folder, recurse)
+            # if none of the above, list all VMs inside the datacenter
+            else:
+                _virtual_machines = get_all_virtual_machines(content, datacenter.vmFolder, recurse)
+        # No datacenter option, list everything. This will be a slow process !
+        else:
+            _virtual_machines = get_all_virtual_machines(content, recurse)
```
Comment (chrrrles) [inline cloud/vmware/vmware_vm_facts.py]: Hmm - I think this would cause a TypeError because two arguments are being passed to a function requiring three.

Comment (ritzk) [inline cloud/vmware/vmware_vm_facts.py]: Thanks, I missed out on this. I have tested, and updated the example to test for these cases
- Gather all vm_facts ( slowest)
  # - Gather all vm_facts form a dc ( not written )
- Gather all vm_facts from a cluster on a dc
- Gather all vm_facts from a folder on dc ( fastest)

**Comment by ritzk** [comment]
version_added has been quoted.

**Comment by gregdek** [comment]
Thanks @ritzk for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ritzk A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ritzk Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ritzk** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @ritzk. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Kicking to ping current maintainers.

**Comment by gregdek** [comment]
Thanks @ritzk. To the current maintainers, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @ritzk for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**